src/tests/ftest/io/NvmeIO.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+import sys
+import json
+import traceback
+import uuid
+import avocado
+
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+import ServerUtils
+import WriteHostFile
+import IorUtils
+
+from GeneralUtils import DaosTestError
+from daos_api import DaosContext, DaosPool, DaosApiError, DaosLog
+
+class NvmeIo(avocado.Test):
+    """
+    Test Class Description:
+        Test the general Metadata operations and boundary conditions.
+    """
+
+    def setUp(self):
+        self.pool = None
+        self.hostlist = None
+        self.hostfile_clients = None
+        self.hostfile = None
+        self.out_queue = None
+        self.pool_connect = True
+
+        with open('../../../.build_vars.json') as json_f:
+            build_paths = json.load(json_f)
+
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+        self.server_group = self.params.get("name", '/server_config/',
+                                            'daos_server')
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+        self.d_log = DaosLog(self.context)
+        self.hostlist = self.params.get("servers", '/run/hosts/*')
+        self.hostfile = WriteHostFile.WriteHostFile(self.hostlist,
+                                                    self.workdir)
+        #Start Server
+        ServerUtils.runServer(self.hostfile, self.server_group, self.basepath)
+
+    def tearDown(self):
+        try:
+            if self.pool_connect:
+                self.pool.disconnect()
+                self.pool.destroy(1)
+        finally:
+            ServerUtils.stopServer(hosts=self.hostlist)
+
+    def verify_pool_size(self, original_pool_info, ior_args):
```

Comment:
So this seems too simple if you can pick different object types, e.g. 3x replication and there is metadata and maybe checksums, etc.

Reply:
Thanks for catching this. I will include replica in size account.